Versioned static manifests

afirth opened the pull request:
Generate static manifests compatible with supported K8s versions, rather than just 1.20

## What this PR does / why we need it:

Currently the helm templates have handling for different kubernetes versions, but the script to generate static manifests uses the helm default version (currently 1.20).
With this change the existing `deploy/static` manifests remain at 1.20 so we don't break docs, but additional folders are underneath with an explicit version.

For more background see #8000 and #7810. #8099 is related.

### Changes
The only real changes here are in [hack/generate-deploy-scripts.sh](https://github.com/kubernetes/ingress-nginx/pull/8162/files#diff-7b3dd6e2019d41d56cf859d2cbae90c0c3957e4e6816fc5ac0df96ef84f75ebfL29-L69)
1. generate for other supported K8s versions
2. don't delete the 1.20 folder (a hack I put in to make #8099 small enough to review)

everything else is the result of running the script.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] New feature (non-breaking change which adds functionality)

## Which issue/s this PR fixes
/fixes #7810

## How Has This Been Tested?
1. Ran `./generate-deploy-scripts.sh`
2. Verified existing manifests updated
3. spot-checked diffs to confirm expected changes present e.g.
```bash
$ diff deploy/static/provider/aws/1.19/deploy.yaml deploy/static/provider/aws/deploy.yaml
```
```diff
0a1
> #GENERATED FOR K8S 1.20
367a369,371
>   ipFamilies:
>   - IPv4
>   ipFamilyPolicy: SingleStack
369c373,374
<   - name: http
---
>   - appProtocol: http
>     name: http
373c378,379
<   - name: https
---
>   - appProtocol: https
>     name: https
398c404,405
<   - name: https-webhook
---
>   - appProtocol: https
>     name: https-webhook
```
1.19 does not have `appProtocol` or `ipFamilies` :+1: 

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- ~~My change requires a change to the documentation.~~
- ~~I have updated the documentation accordingly.~~
(docs updates were completed in #8099)
- [x] I've read the [CONTRIBUTION](https://github.com/kubernetes/ingress-nginx/blob/main/CONTRIBUTING.md) guide
- [n/a] I have added tests to cover my changes.
- [x] All new and existing tests passed.

afirth commented:
/assign @rikatz
9d12bd3 is the script change
6a2a2ce is the manifest generation

boilerplate test broke because I force pushed in the middle but 
/test pull-ingress-nginx-boilerplate
anyway

afirth commented:
/kind feature

iamNoah1 reviewed (commented):
/triage accepted
/priority important-soon

afirth commented:
I updated the description too but understand this could be daunting to a reviewer.
The only real changes here are in hack/generate-deploy-scripts.sh

- generate for other supported K8s versions
- don't delete the 1.20 folder (a hack I put in to make Static manifest generation uses kustomize instead of python #8099 small enough to review)
everything else is the result of running the script.

a reviewer could (untested)
```bash
git checkout main
git checkout 6a2a2ce -- ./hack/generate-deploy-scripts.sh
./hack/generate-deploy-scripts.sh
```
helm, and kustomize must be available (installable with asdf). RELEASE.md was also updated in #8099 with these instructions

longwuyuan commented:
@afirth 
     
<img width="625" alt="image" src="https://user-images.githubusercontent.com/5085914/156134543-f1ec2084-45be-42d8-ad33-b071afe76124.png">

https://kubernetes.slack.com/archives/C021E147ZA4/p1646123996592759?thread_ts=1645961964.408979&cid=C021E147ZA4

afirth commented:
@longwuyuan docs were merged in #8099 https://github.com/kubernetes/ingress-nginx/blob/main/RELEASE.md#e-edit-the-static-manifests to update the python and ruyaml deps to helm+kustomize

afirth commented:
👍 go for it
